drgn/helpers/common/memory.py

```diff
@@ -0,0 +1,88 @@
+# Copyright (c) Meta Platforms, Inc. and affiliates.
+# SPDX-License-Identifier: GPL-3.0-or-later
+
+"""
+Memory
+------
+
+The ``drgn.helpers.memory`` module provides helpers for working with memory and addresses.
+"""
+
+import operator
+from typing import Optional, Union, overload
+
+import drgn
+from drgn import IntegerLike, Object, Program, SymbolKind
+from drgn.helpers.common.format import escape_ascii_string
+from drgn.helpers.linux.slab import find_containing_slab_cache
+
+__all__ = ("identify_address",)
+
+
+_SYMBOL_KIND_STR = {
+    SymbolKind.OBJECT: "object symbol",
+    SymbolKind.FUNC: "function symbol",
+}
+
+
+@overload
+def identify_address(addr: Object) -> Optional[str]:
+    """"""
+    ...
+
+
+@overload
+def identify_address(prog: Program, addr: IntegerLike) -> Optional[str]:
+    """
+    Identify an address (given as a :class:`.Program` and an integer) as a symbol or slab object.
```

Comment:
How about we make this more generic:

```suggestion
    Try to identify what an address refers to.
```

(Yeah I know the original was my wording :stuck_out_tongue:)

Reply:
Yeah I don't claim originality for any of the documentation 😛 Fixed!